src/shared/XLinkDispatcherImpl.c

```diff
@@ -476,20 +476,27 @@ void dispatcherCloseLink(void* fd, int fullClose)
 
     for (int index = 0; index < XLINK_MAX_STREAMS; index++) {
         streamDesc_t* stream = &link->availableStreams[index];
-        if (!stream) {
+        // Wasn't initialized
+        if(stream->id == INVALID_STREAM_ID) {
```

Comment:
Unsafe. All read/writes to the `streamDesc_t` struct must be protected. This line should be inside XLink_sem_wait protection of line PR line 488